src/client/cockpit-client

```diff
@@ -48,6 +49,20 @@ class CockpitClientWindow(Gtk.ApplicationWindow):
         ])
 
 
+    def decide_policy(self, view, decision, decision_type):
+        if decision_type == WebKit2.PolicyDecisionType.NEW_WINDOW_ACTION:
+            uri = decision.get_navigation_action().get_request().get_uri()
+            if uri.startswith('http://127'):
+                logging.error('warning: no support for pop-ups')
+            else:
+                # We can't get the timestamp from the request, so use Gdk.CURRENT_TIME (== 0)
+                Gtk.show_uri_on_window(self, uri, 0)
```

Comment:
Nice! That's half of the fix for https://bugzilla.redhat.com/show_bug.cgi?id=1882677 . I suppose this still won't work in cockpit-desktop as that most likely won't reach out through the ns namespace/flatpak?